Dependencies/AndroidExtensions/Source/JavaWrappers.cpp

```diff
@@ -108,6 +108,9 @@ namespace java::lang
 
     String::operator std::string() const
     {
+        if (m_string == nullptr) {
+            throw std::runtime_error{"Tried to implicitly convert null Java String to C++ String"};
+        }
```

Comment:
Since we don't provide a way to check if the underlying java string is null, would it make sense to provide a conversion operator to `std::optional<std::string>` instead of or in addition to the operator that tries to go straight to `std::string`?